addons/mail/static/src/discuss/gif_picker/common/gif_picker.js

```diff
@@ -166,27 +166,35 @@ export class GifPicker extends Component {
             return;
         }
         this.state.loadingGif = true;
-        try {
-            const params = {
-                country: this.userService.lang.slice(3, 5),
-                locale: this.userService.lang,
-                search_term: this.state.searchTerm,
-            };
-            if (this.next) {
-                params.position = this.next;
-            }
-            const { results, next } = await this.rpc("/discuss/gif/search", params, {
+        const params = {
+            country: this.userService.lang.slice(3, 5),
+            locale: this.userService.lang,
+            search_term: this.state.searchTerm,
+        };
+        if (this.next) {
+            params.position = this.next;
+        }
+        this.sequential(() =>
+            this.rpc("/discuss/gif/search", params, {
                 silent: true,
-            });
-            if (results) {
-                this.next = next;
-                for (const gif of results) {
-                    this.pushGif(gif);
+            })
+        )
+            .then((res) => {
```

Comment:
any particular reason to use `then` syntax rather than `await`?

Reply:
None, I changed it back to try/catch :)